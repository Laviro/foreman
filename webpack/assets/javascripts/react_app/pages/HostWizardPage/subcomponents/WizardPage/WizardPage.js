import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Grid, noop } from 'patternfly-react';
import CardList from '../CardList';
import Filter from '../Filter';

class WizardPage extends Component {
  handleFilter = ({ target: { value } }) => {
    const { updateFilter } = this.props;
    updateFilter(value);
  };

  render() {
    const { title, items, filter } = this.props;
    return (
      <Grid className="foreman_wizard_page">
        <h2>{title}</h2>
        <Filter onChange={this.handleFilter} />
        <CardList items={items} filter={filter} />
      </Grid>
    );
  }
}

WizardPage.propTypes = {
  title: PropTypes.string,
  filter: PropTypes.string,
  items: PropTypes.array,
  updateFilter: PropTypes.func,
};

WizardPage.defaultProps = {
  title: null,
  filter: '',
  items: [],
  updateFilter: noop,
};

export default WizardPage;
