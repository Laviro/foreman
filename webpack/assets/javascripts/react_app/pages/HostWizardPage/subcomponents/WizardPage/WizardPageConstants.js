export const WIZARDPAGE_FILTER_UPDATE = 'WIZARDPAGE_FILTER_UPDATE';
