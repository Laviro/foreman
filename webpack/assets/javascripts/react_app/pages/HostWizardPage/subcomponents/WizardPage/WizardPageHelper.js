export const chunkItemsByAmount = (items, amount = 12) => {
  const chunkedItems = [];
  for (let i = 0; i < items.length; i += amount) {
    chunkedItems.push(items.slice(i, i + amount));
  }
  return chunkedItems;
};
