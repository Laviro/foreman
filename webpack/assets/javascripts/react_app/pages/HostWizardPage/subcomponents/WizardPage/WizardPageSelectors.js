export const selectWizardPage = state => state.wizardPage;
export const selectFilter = state => selectWizardPage(state).filter;
