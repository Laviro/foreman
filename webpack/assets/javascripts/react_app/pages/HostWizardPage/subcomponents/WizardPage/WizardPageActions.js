import { WIZARDPAGE_FILTER_UPDATE } from './WizardPageConstants';

export const updateFilter = filter => ({
  type: WIZARDPAGE_FILTER_UPDATE,
  payload: { filter },
});
