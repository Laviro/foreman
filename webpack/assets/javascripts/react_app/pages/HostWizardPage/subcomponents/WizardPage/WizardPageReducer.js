import Immutable from 'seamless-immutable';

import { WIZARDPAGE_FILTER_UPDATE } from './WizardPageConstants';

const initialState = Immutable({
  filter: '',
});

export default (state = initialState, action) => {
  const { payload } = action;

  switch (action.type) {
    case WIZARDPAGE_FILTER_UPDATE:
      return state.set('filter', payload.filter);
    default:
      return state;
  }
};
