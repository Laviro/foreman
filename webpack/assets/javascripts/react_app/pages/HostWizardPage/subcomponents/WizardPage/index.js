import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './WizardPageActions';
import reducer from './WizardPageReducer';
import { selectFilter } from './WizardPageSelectors';

import WizardPage from './WizardPage';

// map state to props
const mapStateToProps = state => ({
  /** add state keys here */
  filter: selectFilter(state),
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = { wizardPage: reducer };

// export connected component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WizardPage);
