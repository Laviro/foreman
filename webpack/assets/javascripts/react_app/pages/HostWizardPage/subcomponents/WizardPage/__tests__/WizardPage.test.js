import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import WizardPage from '../WizardPage';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('WizardPage', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(WizardPage, fixtures));
});
