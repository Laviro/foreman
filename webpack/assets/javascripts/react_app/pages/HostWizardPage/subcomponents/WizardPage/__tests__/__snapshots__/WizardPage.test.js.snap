// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardPage rendering render without Props 1`] = `
<Grid
  bsClass="container"
  className="foreman_wizard_page"
  componentClass="div"
  fluid={false}
>
  <h2 />
  <Filter
    onChange={[Function]}
  />
  <CardList
    filter=""
    items={Array []}
  />
</Grid>
`;
